xscen/regridding.py

```diff
@@ -52,105 +71,131 @@ def regrid(
+                for d in out.lon.dims:
+                    out[d] = ds_grid[d]
+                    if d not in out.coords:
+                        out = out.assign_coords({d: ds_grid[d]})
+            else:
+                gridmap = np.unique(
+                    [
+                        ds[da].attrs["grid_mapping"]
+                        for da in ds.data_vars
+                        if "grid_mapping" in ds[da].attrs
+                    ]
+                )
+                # Remove the original grid_mapping attribute
+                for v in out.data_vars:
+                    if "grid_mapping" in out[v].attrs:
+                        out[v].attrs.pop("grid_mapping")
+                # Remove the original grid_mapping coordinate if it is still in the output
+                out = out.drop_vars(set(gridmap).intersection(out.variables))
+
+            # History
+            kwargs_for_hist = deepcopy(regridder_kwargs)
+            kwargs_for_hist.setdefault("method", regridder.method)
+            if intermediate_reg_grids and i < len(intermediate_reg_grids):
+                name_inter = list(intermediate_reg_grids.keys())[i]
+                cf_grid_2d_args = intermediate_reg_grids[name_inter]["cf_grid_2d"]
+                new_history = (
+                    f"[{datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S')}] "
+                    f"regridded with regridder arguments {kwargs_for_hist} to a xesmf"
+                    f" cf_grid_2d with arguments {cf_grid_2d_args}  - xESMF v{xe.__version__}"
+                )
+            else:
+                new_history = (
+                    f"[{datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S')}] "
+                    f"regridded with arguments {kwargs_for_hist} - xESMF v{xe.__version__}"
+                )
+            history = (
+                new_history + " \n " + out.attrs["history"]
+                if "history" in out.attrs
+                else new_history
             )
-            # Remove the original grid_mapping attribute
-            for v in out.data_vars:
-                if "grid_mapping" in out[v].attrs:
-                    out[v].attrs.pop("grid_mapping")
-            # Remove the original grid_mapping coordinate if it is still in the output
-            out = out.drop_vars(set(gridmap).intersection(out.variables))
-
-        # History
-        kwargs_for_hist = deepcopy(regridder_kwargs)
-        kwargs_for_hist.setdefault("method", regridder.method)
-        new_history = (
-            f"[{datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S')}] "
-            f"regridded with arguments {kwargs_for_hist} - xESMF v{xe.__version__}"
-        )
-        history = (
-            new_history + " \n " + out.attrs["history"]
-            if "history" in out.attrs
-            else new_history
-        )
-        out.attrs["history"] = history
+            out.attrs["history"] = history
 
+    out = out.drop_vars("latitude_longitude", errors="ignore")
```

Comment:
What's that for?

Reply:
It's a dimension created by cf_grid_2d. I don't think we want it to appear in the output dataset.

Reply:
Ah, gotcha! I didn't know where that was coming from. I agree that this should be removed.